### ISSUE TITLE
[UX] Config page: Add icons to the config sections.

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#1178]
Here's how this page looks with the Seven admin theme...

D7:

![d7-config_page-seven_theme](https://cloud.githubusercontent.com/assets/2423362/8392059/55522064-1d20-11e5-89b1-410842ffc632.png)

D8:

![d8-config_page-seven_theme](https://cloud.githubusercontent.com/assets/2423362/8392063/611f0376-1d20-11e5-92d4-348cca8fb6a8.png)

Backdrop:

![bd-config_page-seven_theme](https://cloud.githubusercontent.com/assets/2423362/8392065/6d8c60ae-1d20-11e5-9aa9-5b66be1933b7.png)

...despite the minor changes, all same all. Here's how the same page looks with the [Adminimal](https://www.drupal.org/project/adminimal_theme) theme (my theme of choice for the backend of my sites):

![d7-config_page-adminimal_theme](https://cloud.githubusercontent.com/assets/2423362/8392081/c31e7552-1d20-11e5-8ecf-2fd2f450047c.png)

[Rubik](https://www.drupal.org/project/rubik) is another admin theme that makes use of category icons for the config page among other admin pages. Also notice the search filter (#978) as well as the hide/show descriptions toggle (#979) (though none of them are "instant" - you have to submit the form):

![d7-config_page-rubik_theme](https://cloud.githubusercontent.com/assets/2423362/8392112/17c03d10-1d22-11e5-9820-55b5e9695853.png)

---
~PR by @wesruv: https://github.com/backdrop/backdrop/pull/1074~
PR by @jenlampton: https://github.com/backdrop/backdrop/pull/1967

**Comment by klonos** [comment]
> ...the way I navigate this page currently is by knowing that the performance settings are half way down on the right hand side. Let's make sure that if we break these habits for people we have a good reason, and a better alternative order :)

I think that the addition of icons in the Adminimal theme does not require people to "know" where to find things - it allows finding them with a visual quick scan. Of course #978 would also greatly help to that end.

**Comment by jenlampton** [comment]
I do actually like this idea. 
But what happens when contrib adds 100 new sections to this page, will we provide a default icon that can be used for all "unknown" sections?

**Comment by klonos** [comment]
> ...will we provide a default icon that can be used for all "unknown" sections?

Yeah, I guess. The ideal would be to allow modules to define their own icon for new categories and if they don't, then we should fall-back to a generic one.

**Comment by wesruv** [comment]
@jenlampton @klonos 

That's the way I've coded it in my branch, in this screenshot I removed the style for the system icon so it falls back to the default:
![image](https://cloud.githubusercontent.com/assets/5607236/9948591/76370ca8-5d72-11e5-8c32-8e2b255d8e6b.png)

**Comment by klonos** [comment]
Already looks good. Should we also add the bigger, faded-out versions of the icons in the sections backgrounds too or do you guys think we'd over-complicate the UI? I personally think they are cool.

**Comment by wesruv** [comment]
I don't think that jives with our minimalist look, I vote nay on watermarks.

**Comment by docwilmot** [comment]
Just curious, is this the start of the Icons Everywhere Initiative (IEI)? Or is this just for that admin page? 

P.S. Let it be widely known that I came up with the IEI acronym. Trademarked.

**Comment by wesruv** [comment]
lol, I think icons can be useful in some circumstances :P

Not opposed to other places as long as there's a good reason.

**Comment by wesruv** [comment]
Here's my current implementation, which has a PR here:
https://github.com/backdrop/backdrop/pull/1074

### Configuration page
![configuration l backdropdev com](https://cloud.githubusercontent.com/assets/5607236/9841549/1b33cfc2-5a6f-11e5-854c-c656e34fb314.png)

### Performance
![performance l backdropdev com](https://cloud.githubusercontent.com/assets/5607236/9841560/3d8fe97a-5a6f-11e5-9755-524df5dafbf6.png)

### Permissions
![permissions l backdropdev com](https://cloud.githubusercontent.com/assets/5607236/9841567/46f60a94-5a6f-11e5-969a-112e4fe406e9.png)

### Administration Bar
![administration bar l backdropdev com](https://cloud.githubusercontent.com/assets/5607236/9841602/c3f3e1ba-5a6f-11e5-9c2d-c2feab409419.png)

**Comment by wesruv** [comment]
Updated with original explanation from #506

**Comment by quicksketch** [comment]
I think this looks great and is a good step forward for the (rather humble) aesthetic of our admin section.

Wes filed a good looking PR at https://github.com/backdrop/backdrop/pull/1074. But one question I couldn't help thinking: SVG?

I know we went over all this in #364. But if it's possible to do this as SVG with PNG fallbacks (as I think we settled on back then), it would be more future-proof and look a lot better on HiDPI screens.

**Comment by wesruv** [comment]
@quicksketch We'll need to do a feature test with JS to get that to work, but that won't be a problem.

**Comment by wesruv** [comment]
Lol, remember this? I'll try and get this tied up soon... 

/me knocks on wood.

**Comment by jenlampton** [comment]
Omg, I love how this looks!  Also the PR looks close enough that maybe we could wrap this up pretty quickly - though it will need quite a rebase at this point :)

> But if it's possible to do this as SVG with PNG fallbacks (as I think we settled on back then), it would be more future-proof and look a lot better on HiDPI screens.

All the other icons we added to core in recent versions are PNG versions of the font-awesome icons, as generated by http://fa2png.io/.  I think we should do the same for these, and stay away from SVG until they become more widely adopted.

**Comment by jenlampton** [comment]
I pushed a new PR, with a rebase, and the following changes:

* I replaced all the icons with more up-to-date ones from the font-awesome library, as generated by http://fa2png.io.
* All icons are .pngs at 64px, resized via CSS to the size we need on the config page. This will make them look good on retina displays, and if we decide to add larger versions of the icons into the page-headings on all admin pages, they are large enough to still use the css-size-down method there as well.
* I named the new icons with their font-awesome name, including the `fa-`. My thought process was: that will make it easy for anyone looking at the icon files to know they are font-awesome icons. 
* All new icons are Black, to match the font in the heading text on the config page. 
* I removed the icons from the page titles (for now).  I love how the page titles look with icons, but there are other admin sections (outside "Configuration") that we would need to worry about too. 

Ready for a new review.

**Comment by jenlampton** [comment]
Latest PR seems to be working. Question there about what's allowed in 1.x, we may need to leave the existing selectors alone.

**Comment by quicksketch** [comment]
Updating the selectors to be less specific is always a good thing. I think the CSS changes are fine. I left a review on the issue, this looks pretty close!

**Comment by olafgrabienski** [comment]
I like this new look of the configuration overview page. The icons help a lot, also because the items on the page are ordered in a strange way.

Off topic regarding the ordering of items on the config page (see the actual order below): Does anybody know what's the idea behind it, or if there was already filed an issue about it?

```
Left:
- User Accounts
- Media
- Search
- Regional and Language
- Administration
- Content Authoring

Right:
- System
- URL Handling
- Development
- Web services
```

**Comment by jenlampton** [comment]
We mentioned the strange order in https://github.com/backdrop/backdrop-issues/issues/506 but I'm not sure there's a sub-issue to address it independently, though we could probably fix that pretty quickly.

IIRC the order was supposed to be set manually to get the most important items at the top of the page, to avoid scrolling. I'm not sure that's worked and I agree it does make the page harder to navigate if you are looking for something in particular. I, for one, would be happy to return to alpha order.

**Comment by jenlampton** [comment]
Pushed a new PR with the suggested changes:
<img width="1276" alt="screen shot 2017-08-31 at 4 13 45 pm" src="https://user-images.githubusercontent.com/397895/29949130-7bd590b2-8e67-11e7-849b-1083de5e5374.png">

**Comment by quicksketch** [comment]
The PR is causing some test failures due to PHP notices: https://github.com/backdrop/backdrop/pull/1967#pullrequestreview-60030382

Naming and CSS changes all look great. One minor item that seems unnecessary is the strict matching of Font Awesome's file names. e.g.

```
hdd-o--black--64.png
pencil-square-o--black--64.png
picture-o--black--64.png
```

I'm assuming the `-o` is for "open"? Perhaps Font Awesome has non-open versions of these icons as well that it needs to differentiate? The names are probably fine if we add a line to the CSS file indicating that the file names line up with Font Awesome. Either that or we shorten the name to remove the unnecessary `-o`.

Also related to attribution, Font Awesome is GPL compatible but I believe we need to include license information. Per our README.md in the root, any place where we have assets under a difference license, we include license information where those libraries or files exist. I would suggest we make a `README.md` file in the `seven` theme directory (as Basis does) and include the License information for Font Awesome in there. Something along the lines of:

```
## License
Seven theme is [GPL v2](http://www.gnu.org/licenses/gpl-2.0.html) (or higher)
software. See the LICENSE.txt file in the /core directory for complete text.
Distributions of this software may relicense it as any later version of the GPL.

Seven theme includes icons derived from [Font Awesome](http://fontawesome.io/),
which is licensed under the [SIL OFL 1.1](http://scripts.sil.org/OFL).
```

**Comment by olafgrabienski** [comment]
@jenlampton Thanks for pointing out the discussion about the order of the Configuration page items in the meta issue #506. There was no sub-issue filed about the ordering questions, but they are listed in the issue description:

> The categories seem to be in random order. They should be in alphabetical order (unless somebody has another suggestion).
>  The items under each category seem to be randomly placed as well. They should be listed alphabetically (unless somebody has another suggestion).

Most of the discussion in the comments is about ordering and the (not yet responsive) two column layout but also about icons (which led to *this* issue). It's hard to summarize the discussion, there were however good and easy to implement suggestions by @klonos and @docwilmot to fix the ordering alphabetically and to make the page responsive at the same time. At a certain point, before someone made a PR for the alphabetical ordering and a reasonable responsive layout, the issue moved focus to other questions, although a fix seemed pretty near. I'm not sure how to proceed now: open a sub-issue and copy the discussion there?

**Comment by jenlampton** [comment]
Yes, good idea. I think a sub issue would help us stay focused on the
ordering.

**Comment by quicksketch** [comment]
> This is a reasonable solution, though it wouldn't be necessary if the file names also began with `fa-` ;) Where should we put it? Right under `@file` at the top?

I'd put them right before we start defining all the background images. Or if you want, put it in the new README.md file instead.

**Comment by jenlampton** [comment]
Okay, I added a README.md file into the root of the theme with licensing info for the new icons, and pushed a new PR: https://github.com/backdrop/backdrop/pull/1967

Also, here's a nice little complementary fix for the admin/config page: https://github.com/backdrop/backdrop-issues/issues/2828

**Comment by olafgrabienski** [comment]
@jenlampton Thanks for making the sub issue ... and fixing it!

**Comment by quicksketch** [comment]
The lastest PR looks just great. Nice job on the README.md file. I've merged https://github.com/backdrop/backdrop/pull/1967 into 1.x for 1.8.0.
